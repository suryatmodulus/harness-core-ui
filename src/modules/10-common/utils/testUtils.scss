/*
 * Copyright 2021 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

.browser {
  border: 1px solid #cccccc;
  border-radius: 8px;
  min-height: 300px;
  overflow: auto;

  .browser-header {
    background-color: #cccccc;
    padding: 8px 16px;
    display: flex;
  }

  .browser-path {
    display: block;
    border: none;
    line-height: 14px;
    font-size: 12px;
    color: #444444;
    border-radius: 9px;
    min-width: 100px;
    padding: 2px 8px;
    flex: 1 1 auto;
    border: 1px solid transparent;

    &:focus {
      border-color: rgb(30, 167, 253);
      outline: none;
    }
  }

  .browser-content {
    padding: 8px;
  }
}
