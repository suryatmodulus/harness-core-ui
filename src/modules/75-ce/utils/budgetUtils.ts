/*
 * Copyright 2021 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

export enum AlertStatus {
  Danger,
  Warning,
  Info
}

interface CostStatusInfo {
  cost: number
  ratio: number
  percentage: number
  status: AlertStatus
}

export interface getAllBudgetCostInfoReturnType {
  actualCostStatus: CostStatusInfo
  forecastedCostStatus: CostStatusInfo
  budgetAmountStatus: CostStatusInfo
}

export const getAllBudgetCostInfo: (
  forecastedCost: number,
  actualCost: number,
  budgetAmount: number
) => getAllBudgetCostInfoReturnType = (forecastedCost, actualCost, budgetAmount) => {
  /**
   * Find out the max cost of all and it will be considered as 100%
   * and the rest of the cost percentages will be calculated based on that
   */

  const baseCost = Math.max(actualCost, forecastedCost, budgetAmount)
  const actualCostRatio = actualCost / baseCost
  const forecastCostRatio = forecastedCost / baseCost
  const budgetAmountRatio = budgetAmount / baseCost

  const actualCostPerWithBudget = (actualCost / budgetAmount) * 100
  const forecastCostPerWithBudget = (forecastedCost / budgetAmount) * 100

  const getAlertStatus: (percentage: number) => AlertStatus = percentage => {
    if (percentage > 100) {
      return AlertStatus.Danger
    }
    if (percentage > 90) {
      return AlertStatus.Warning
    }
    return AlertStatus.Info
  }

  const actualCostStatus: CostStatusInfo = {
    cost: actualCost,
    ratio: actualCostRatio,
    percentage: actualCostPerWithBudget,
    status: getAlertStatus(actualCostPerWithBudget)
  }
  const forecastedCostStatus: CostStatusInfo = {
    cost: forecastedCost,
    ratio: forecastCostRatio,
    percentage: forecastCostPerWithBudget,
    status: getAlertStatus(forecastCostPerWithBudget)
  }
  const budgetAmountStatus: CostStatusInfo = {
    cost: budgetAmount,
    ratio: budgetAmountRatio,
    percentage: 100,
    status: AlertStatus.Info
  }

  return {
    actualCostStatus,
    forecastedCostStatus,
    budgetAmountStatus
  }
}
