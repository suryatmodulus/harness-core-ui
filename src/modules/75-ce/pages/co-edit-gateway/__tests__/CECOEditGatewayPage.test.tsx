/*
 * Copyright 2021 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

import React from 'react'
import { render } from '@testing-library/react'
import { TestWrapper } from '@common/utils/testUtils'
import CECOEditGatewayPage from '../CECOEditGatewayPage'

const mockedService = {
  id: 224,
  name: 'k8s rule edit test',
  fulfilment: 'ondemand',
  kind: 'k8s',
  org_id: 'lwdev1',
  project_id: 'Rishabh_bugbash2303',
  cloud_account_id: 'CeDevShaAWS',
  idle_time_mins: 15,
  host_name: 'lwdev1-k8s-rule-edit-test.gateway.lightwing.io',
  custom_domains: null,
  match_all_subdomains: false,
  disabled: false,
  created_by: 'lv0euRhKRCyiXWzS7pOg6g',
  routing: {
    instance: null,
    lb: null,
    k8s: {
      RuleJson:
        '{"apiVersion":"lightwing.lightwing.io/v1","kind":"AutoStoppingRule","metadata":{"name":"k8s-rule-edit-test","annotations":{"harness.io/project-id":"Rishabh_bugbash2303","harness.io/org-id":"lwdev1","harness.io/cloud-connector-id":"CeDevShaAWS","nginx.ingress.kubernetes.io/configuration-snippet":"more_set_input_headers \\"AutoStoppingRule: lwdev1-Rishabh_bugbash2303-k8s-rule-edit-test\\""}},"spec":{"idleTimeMins":15,"rules":[{"host":"randomtest.lw.in","http":{"paths":[{"path":"/","pathType":"Prefix","backend":{"service":{"name":"test-service","port":{"number":80}}}}]}}]}}'
    },
    ports: []
  },
  health_check: {
    protocol: 'http',
    path: '/',
    port: 80,
    timeout: 30,
    id: ''
  },
  opts: {
    preserve_private_ip: false,
    delete_cloud_resources: false,
    always_use_private_ip: false
  },
  metadata: {
    custom_domain_providers: null,
    target_group_details: null,
    access_details: {
      backgroundTasks: {
        selected: false
      },
      dnsLink: {
        selected: false
      },
      ipaddress: {
        selected: false
      },
      rdp: {
        selected: false
      },
      ssh: {
        selected: false
      }
    },
    cloud_provider_details: {
      name: 'CeDevShaAWS'
    },
    service_errors: null,
    kubernetes_connector_id: 'ry_kops_test_1',
    health_check_details: null
  },
  access_point_id: null,
  status: 'created',
  created_at: '2021-07-01T06:39:20.845693Z',
  updated_at: '2021-07-01T07:47:15.527434Z',
  account_identifier: 'kmpySmUISimoRrJL6NL73w'
}

const mockedData = { response: [] }

const mockedInstances = [
  {
    id: 'id',
    ipv4: '',
    launch_time: '', // eslint-disable-line
    name: 'name',
    region: 'us-west-2',
    status: 'running',
    tags: '',
    type: 'type',
    vpc: 'vpcid'
  }
]

const mockedAsg = {
  id: 'mockAsg',
  desired: 0,
  max: 3,
  min: 0,
  spot: 0,
  on_demand: 1, // eslint-disable-line
  provider_name: '', // eslint-disable-line
  // eslint-disable-next-line
  target_groups: [
    {
      id: 'target',
      name: 'my-targets',
      port: 80,
      protocol: 'HTTP',
      vpc: 'vpc'
    }
  ],
  availability_zones: ['us-east-1a'], // eslint-disable-line
  status: '',
  region: 'us-east-1',
  meta: {
    // eslint-disable-next-line
    mixed_instance_policy: {
      InstancesDistribution: {
        OnDemandAllocationStrategy: 'prioritized',
        OnDemandBaseCapacity: 1,
        OnDemandPercentageAboveBaseCapacity: 0,
        SpotAllocationStrategy: 'lowest-price',
        SpotInstancePools: 2,
        SpotMaxPrice: null
      },
      LaunchTemplate: {
        LaunchTemplateSpecification: {
          LaunchTemplateId: 'lt-03',
          LaunchTemplateName: 'lwcg',
          Version: null
        },
        Overrides: [{ InstanceType: '', LaunchTemplateSpecification: null, WeightedCapacity: null }]
      }
    }
  },
  mixed_instance: true // eslint-disable-line
}

const mockedStaticSchedulesList = { response: [] }

const mockedRouteDetails = { response: { service: mockedService } }

const testPath = '/account/:accountId/ce/orgs/:orgIdentifier/projects/:projectIdentifier/autostopping-rules/edit/10'

const testParams = { accountId: 'accountId', orgIdentifier: 'orgIdentifier', projectIdentifier: 'projectIdentifier' }

jest.mock('@common/components/YAMLBuilder/YamlBuilder')

jest.mock('services/lw', () => ({
  useRouteDetails: jest.fn().mockImplementation(() => ({ data: mockedRouteDetails, loading: false })),
  useAllServiceResources: jest.fn().mockImplementation(() => ({ data: mockedData, loading: false })),
  useSaveService: jest.fn().mockImplementation(() => ({
    mutate: jest.fn()
  })),
  useAllResourcesOfAccount: jest.fn().mockImplementation(() => ({
    mutate: jest.fn(() =>
      Promise.resolve({
        response: mockedInstances
      })
    ),
    loading: false
  })),
  useGetAllASGs: jest
    .fn()
    .mockImplementation(() => ({ mutate: jest.fn(() => Promise.resolve({ response: [mockedAsg] })), loading: false })),
  useSecurityGroupsOfInstances: jest.fn().mockImplementation(() => ({
    mutate: jest.fn(() => Promise.resolve({ response: {} })),
    loading: false
  })),
  useGetServices: jest.fn().mockImplementation(() => ({ data: mockedData, loading: false, error: null })),
  useListStaticSchedules: jest.fn().mockImplementation(() => ({
    data: mockedStaticSchedulesList,
    loading: false,
    refetch: jest.fn(() => Promise.resolve({ response: null }))
  })),
  useCreateStaticSchedules: jest.fn().mockImplementation(() => ({
    mutate: jest.fn()
  })),
  useDeleteStaticSchedule: jest.fn().mockImplementation(() => ({
    mutate: jest.fn()
  }))
}))

describe('Edit rule page', () => {
  test('render edit rule page successfully', () => {
    const { container } = render(
      <TestWrapper path={testPath} pathParams={testParams}>
        <CECOEditGatewayPage />
      </TestWrapper>
    )

    expect(container).toMatchSnapshot()
  })
})
