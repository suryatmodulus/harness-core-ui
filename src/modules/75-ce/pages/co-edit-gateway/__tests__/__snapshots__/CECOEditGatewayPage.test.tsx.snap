// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Edit rule page render edit rule page successfully 1`] = `
<div>
  <div
    class="StyledProps--font StyledProps--main"
    style="overflow: scroll; max-height: 100vh;"
  >
    <div
      class="breadCrumb breadcrumbs"
    >
      <div>
        <a
          class="breadcrumb"
          href="/account/accountId/ce/autostopping-rules"
        >
          ce.co.breadCrumb.rules
        </a>
        <span
          class="bp3-icon StyledProps--main separator"
          data-icon="main-chevron-right"
        >
          <svg
            height="11"
            viewBox="0 0 8 14"
            width="11"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M7.7 6.3l-6-6C1.5.1 1.3 0 1 0 .4 0 0 .4 0 1c0 .3.1.5.3.7L5.6 7 .3 12.3c-.2.2-.3.4-.3.7 0 .6.4 1 1 1 .3 0 .5-.1.7-.3l6-6c.2-.2.3-.4.3-.7s-.1-.5-.3-.7z"
              fill="#4d5969"
            />
          </svg>
        </span>
      </div>
      <div>
        <span
          class="lastBreadcrumb"
        >
          k8s rule edit test
        </span>
      </div>
    </div>
    <div
      class="StyledProps--font StyledProps--main detailsTab"
    >
      <div
        class="bp3-tabs Tabs--main Tabs--horizontal"
      >
        <div
          class="bp3-tab-list"
          role="tablist"
        >
          <div
            class="bp3-tab-indicator-wrapper"
            style="height: 0px; transform: translateX(0px) translateY(0px); width: 0px;"
          >
            <div
              class="bp3-tab-indicator"
            />
          </div>
          <div
            aria-controls="bp3-tab-panel_tabsId1_configuration"
            aria-disabled="true"
            aria-expanded="true"
            aria-selected="true"
            class="bp3-tab"
            data-tab-id="configuration"
            id="bp3-tab-title_tabsId1_configuration"
            role="tab"
          >
            <div
              class="Layout--horizontal StyledProps--main"
            >
              <span
                class="bp3-icon bp3-icon-tick-circle StyledProps--main greenSymbol"
                icon="tick-circle"
              >
                <svg
                  data-icon="tick-circle"
                  height="16"
                  viewBox="0 0 16 16"
                  width="16"
                >
                  <desc>
                    tick-circle
                  </desc>
                  <path
                    d="M8 16c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8zm4-11c-.28 0-.53.11-.71.29L7 9.59l-2.29-2.3a1.003 1.003 0 00-1.42 1.42l3 3c.18.18.43.29.71.29s.53-.11.71-.29l5-5A1.003 1.003 0 0012 5z"
                    fill-rule="evenodd"
                  />
                </svg>
              </span>
              <p
                class="StyledProps--font StyledProps--main tabTitle"
              >
                1. 
                configuration
              </p>
            </div>
          </div>
          <div
            aria-controls="bp3-tab-panel_tabsId1_setupAccess"
            aria-disabled="true"
            aria-expanded="false"
            aria-selected="false"
            class="bp3-tab"
            data-tab-id="setupAccess"
            id="bp3-tab-title_tabsId1_setupAccess"
            role="tab"
          >
            <div
              class="Layout--horizontal StyledProps--main"
            >
              <span
                class="bp3-icon bp3-icon-symbol-circle StyledProps--main symbol"
                icon="symbol-circle"
              >
                <svg
                  data-icon="symbol-circle"
                  height="16"
                  viewBox="0 0 16 16"
                  width="16"
                >
                  <desc>
                    symbol-circle
                  </desc>
                  <path
                    d="M8 3.01a5 5 0 100 10 5 5 0 100-10z"
                    fill-rule="evenodd"
                  />
                </svg>
              </span>
              <p
                class="StyledProps--font StyledProps--main tabTitle"
              >
                2. 
                ce.co.autoStoppingRule.setupAccess.pageName
              </p>
            </div>
          </div>
          <div
            aria-controls="bp3-tab-panel_tabsId1_review"
            aria-disabled="true"
            aria-expanded="false"
            aria-selected="false"
            class="bp3-tab"
            data-tab-id="review"
            id="bp3-tab-title_tabsId1_review"
            role="tab"
          >
            <div
              class="Layout--horizontal StyledProps--main"
            >
              <span
                class="bp3-icon bp3-icon-symbol-circle StyledProps--main symbol"
                icon="symbol-circle"
              >
                <svg
                  data-icon="symbol-circle"
                  height="16"
                  viewBox="0 0 16 16"
                  width="16"
                >
                  <desc>
                    symbol-circle
                  </desc>
                  <path
                    d="M8 3.01a5 5 0 100 10 5 5 0 100-10z"
                    fill-rule="evenodd"
                  />
                </svg>
              </span>
              <p
                class="StyledProps--font StyledProps--main tabTitle"
              >
                3. 
                review
              </p>
            </div>
          </div>
        </div>
        <div
          aria-hidden="false"
          aria-labelledby="bp3-tab-title_tabsId1_configuration"
          class="bp3-tab-panel"
          id="bp3-tab-panel_tabsId1_configuration"
          role="tabpanel"
        >
          <div
            class="Layout--vertical StyledProps--main page"
          >
            <div
              class="StyledProps--font StyledProps--main"
              style="padding-top: 10px;"
            >
              <div
                class="Layout--vertical Layout--layout-spacing-large StyledProps--main StyledProps--padding-large"
              >
                <div
                  class="Layout--vertical Layout--layout-spacing-small StyledProps--main configStepContainer"
                  id="configStep1"
                >
                  <div
                    class="Layout--horizontal StyledProps--main stepInfo"
                  >
                    <span
                      style="color: rgb(30, 92, 31); background-color: rgb(228, 247, 225);"
                    >
                      Step 1 of 3
                    </span>
                  </div>
                  <p
                    class="StyledProps--font StyledProps--main title"
                    data-tooltip-id="defineAwsAsRule"
                  >
                    ce.co.autoStoppingRule.configuration.step1.title
                  </p>
                  <p
                    class="StyledProps--font StyledProps--main subTitle"
                  >
                    ce.co.autoStoppingRule.configuration.step1.subTitle
                  </p>
                  <div
                    class="childrenContainer"
                  >
                    <div
                      class="Layout--horizontal StyledProps--main"
                    >
                      <div
                        class="bp3-card bp3-elevation-0 Card--card displayCard"
                      >
                        <span
                          class="bp3-icon StyledProps--main"
                          data-icon="service-aws"
                        >
                          <svg
                            height="30"
                            viewBox="0 0 14 10"
                            width="30"
                            xmlns="http://www.w3.org/2000/svg"
                          >
                            <path
                              d="M4 3.9v.4c0 .1.1.2.1.3v.1c0 .1 0 .1-.1.1l-.1.2h-.1s-.1 0-.1-.1c-.1.1-.2 0-.2-.1s-.1-.2-.1-.3c-.4.5-.9.7-1.4.7-.4 0-.7-.1-.9-.3S.7 4.3.7 4c0-.4.1-.7.4-.9s.6-.4 1.1-.4h.5c.2 0 .3.1.5.1v-.3c0-.3-.1-.6-.2-.7-.1-.2-.3-.2-.7-.2-.2 0-.3 0-.5.1-.2 0-.3.1-.5.2 0 0-.1 0-.2.1H1v-.5-.2l.1-.1c.3-.1.4-.1.6-.2.3 0 .5-.1.8-.1.5 0 .9.1 1.2.4s.3.6.3 1.1zm-1.8.7c.2 0 .3 0 .5-.1.1-.1.3-.2.4-.3s.1-.2.1-.3V3.3c-.1 0-.2 0-.3-.1h-.4c-.3 0-.5.1-.7.2s-.3.3-.3.5.1.4.2.5.3.2.5.2zm3.6.5h-.2s-.1-.1-.1-.2l-1-3.6v-.2c0-.1 0-.1.1-.1H5.2s.1.1.1.2l.8 3 .7-3c0-.1 0-.1.1-.2h.8s.1.1.1.2l.7 3 .8-3c0-.1 0-.1.1-.2H10c.1 0 .1 0 .1.1v.2L8.8 4.8c0 .1 0 .2-.1.2h-.5c-.1.1-.2.1-.2 0 0 0-.1-.1-.1-.2l-.7-2.9-.7 2.9c0 .1 0 .1-.1.2h-.2zm5.9.1c-.2 0-.5 0-.7-.1s-.4-.1-.5-.2c-.1 0-.1-.1-.1-.1v-.4c0-.1 0-.1.1-.1h.2c.2.1.3.1.5.2.2 0 .4.1.5.1.3 0 .5 0 .7-.1.1-.1.2-.3.2-.4s0-.2-.1-.3-.2-.2-.5-.2l-.7-.2c-.3-.1-.6-.3-.7-.5s-.2-.4-.2-.7c0-.2 0-.3.1-.5.1-.1.2-.3.3-.4s.3-.2.5-.2c.2-.1.4-.1.6-.1h.6c.1 0 .2 0 .3.1.1 0 .1 0 .2.1.1 0 .1.1.1.1V1.6c0 .1 0 .1-.1.1s-.1 0-.2-.1c-.3-.1-.6-.2-.9-.2s-.5 0-.6.1-.2.2-.2.4c0 .1 0 .2.1.3s.3.2.5.2l.6.2c.3.1.6.2.7.4s.2.4.2.6 0 .4-.1.5c-.1.2-.2.3-.3.4-.1.5-.3.5-.5.6s-.4.1-.6.1z"
                              fill="#252f3e"
                            />
                            <g
                              fill="#f90"
                            >
                              <path
                                d="M12.5 7.4C11 8.5 8.9 9.1 7 9.1c-2.6 0-5-1-6.7-2.6-.1-.1 0-.3.2-.2 1.9 1.1 4.3 1.8 6.7 1.8 1.8 0 3.5-.4 5.1-1.1.2-.1.5.2.2.4z"
                              />
                              <path
                                d="M13.1 6.7c-.2-.2-1.3-.1-1.7-.1-.1 0-.2-.1 0-.2.9-.6 2.3-.4 2.4-.2.2.2 0 1.6-.8 2.3-.1.1-.2 0-.2-.1.1-.5.5-1.5.3-1.7z"
                              />
                            </g>
                          </svg>
                        </span>
                        <p
                          class="StyledProps--font StyledProps--main StyledProps--font-small"
                          style="margin-top: 5px;"
                        >
                          AWS
                        </p>
                      </div>
                      <div
                        class="OverlaySpinner--overlaySpinner"
                      >
                        <form
                          class="FormikForm--main step1Form"
                        >
                          <div>
                            <div
                              class="Layout--horizontal Layout--layout-spacing-xxxlarge StyledProps--main"
                            >
                              <div
                                class="Layout--vertical StyledProps--main formElement"
                              >
                                <p
                                  class="StyledProps--font StyledProps--main"
                                  style="font-size: 16px; font-weight: 500; color: rgb(11, 11, 13);"
                                >
                                  AutoStopping Rule Name
                                </p>
                                <div
                                  class="bp3-form-group bp3-disabled"
                                  data-id="gatewayName-0"
                                >
                                  <label
                                    class="bp3-label"
                                    for="gatewayName"
                                  >
                                    <span
                                      class="Tooltip--acenter"
                                      data-tooltip-id="coGatewayConfig_gatewayName"
                                    >
                                      ce.co.gatewayConfig.name
                                    </span>
                                     
                                    <span
                                      class="bp3-text-muted"
                                    />
                                  </label>
                                  <div
                                    class="bp3-form-content"
                                  >
                                    <div
                                      class="bp3-input-group bp3-disabled"
                                    >
                                      <input
                                        autocomplete="off"
                                        class="bp3-input"
                                        disabled=""
                                        name="gatewayName"
                                        placeholder="ce.co.autoStoppingRule.configuration.step1.nameInputPlaceholder"
                                        type="text"
                                        value="k8s rule edit test"
                                      />
                                    </div>
                                  </div>
                                </div>
                              </div>
                              <div
                                class="Layout--vertical StyledProps--main formElement"
                              >
                                <p
                                  class="StyledProps--font StyledProps--main"
                                  style="font-size: 16px; font-weight: 500; color: rgb(11, 11, 13);"
                                >
                                  Idle Time (mins)
                                  <span
                                    class="bp3-icon StyledProps--main"
                                    data-icon="info"
                                  >
                                    <svg
                                      height="16"
                                      viewBox="0 0 12 12"
                                      width="16"
                                      xmlns="http://www.w3.org/2000/svg"
                                    >
                                      <path
                                        clip-rule="evenodd"
                                        d="M0 6a6 6 0 1012 0A6 6 0 000 6zm.75 0a5.25 5.25 0 1010.5 0A5.25 5.25 0 00.75 6zm4.206 3.104v.646h2.682v-.646l-.836-.144V4.203H4.956v.65l.836.144V8.96l-.836.144zm1.846-7.352v1.03h-1.01v-1.03h1.01z"
                                        fill-rule="evenodd"
                                      />
                                    </svg>
                                  </span>
                                </p>
                                <div
                                  class="bp3-form-group"
                                  data-id="idleTime-1"
                                >
                                  <label
                                    class="bp3-label"
                                    for="idleTime"
                                  >
                                    <span
                                      class="Tooltip--acenter"
                                      data-tooltip-id="coGatewayConfig_idleTime"
                                    >
                                      <div
                                        class="Layout--horizontal Layout--layout-spacing-small StyledProps--main"
                                      >
                                        <p
                                          class="StyledProps--font StyledProps--main"
                                          style="font-size: 13px;"
                                        >
                                          ce.co.autoStoppingRule.configuration.step1.form.idleTime.label
                                        </p>
                                      </div>
                                    </span>
                                     
                                    <span
                                      class="bp3-text-muted"
                                    />
                                  </label>
                                  <div
                                    class="bp3-form-content"
                                  >
                                    <div
                                      class="bp3-input-group"
                                    >
                                      <input
                                        autocomplete="off"
                                        class="bp3-input"
                                        name="idleTime"
                                        placeholder="ce.co.autoStoppingRule.configuration.step1.idleTimeInputPlaceholder"
                                        type="text"
                                        value="15"
                                      />
                                    </div>
                                  </div>
                                </div>
                              </div>
                            </div>
                          </div>
                        </form>
                      </div>
                    </div>
                  </div>
                </div>
                <div
                  class="Layout--vertical Layout--layout-spacing-small StyledProps--main configStepContainer"
                  id="configStep2"
                >
                  <div
                    class="Layout--horizontal StyledProps--main stepInfo"
                  >
                    <span
                      style="color: rgb(2, 120, 213); background-color: rgb(239, 248, 254);"
                    >
                      Step 2 of 3
                    </span>
                  </div>
                  <p
                    class="StyledProps--font StyledProps--main title"
                  >
                    ce.co.autoStoppingRule.configuration.step2.title
                    <span
                      class="bp3-icon StyledProps--main"
                      data-icon="info"
                    >
                      <svg
                        height="16"
                        viewBox="0 0 12 12"
                        width="16"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <path
                          clip-rule="evenodd"
                          d="M0 6a6 6 0 1012 0A6 6 0 000 6zm.75 0a5.25 5.25 0 1010.5 0A5.25 5.25 0 00.75 6zm4.206 3.104v.646h2.682v-.646l-.836-.144V4.203H4.956v.65l.836.144V8.96l-.836.144zm1.846-7.352v1.03h-1.01v-1.03h1.01z"
                          fill-rule="evenodd"
                        />
                      </svg>
                    </span>
                  </p>
                  <p
                    class="StyledProps--font StyledProps--main subTitle"
                  >
                    ce.co.autoStoppingRule.configuration.step2.subTitle
                  </p>
                  <div
                    class="childrenContainer"
                  >
                    <div
                      class="Layout--vertical StyledProps--main step2Container"
                    >
                      <p
                        class="StyledProps--font StyledProps--main radioGroupTitle"
                      >
                        Select the resources to be managed by the rule
                      </p>
                      <div
                        class="radioGroup"
                      >
                        <label
                          class="bp3-control bp3-radio"
                        >
                          <input
                            name="Blueprint3.RadioGroup-0"
                            type="radio"
                            value="INSTANCES"
                          />
                          <span
                            class="bp3-control-indicator"
                          />
                          EC2 VM(s)
                        </label>
                        <label
                          class="bp3-control bp3-radio"
                        >
                          <input
                            name="Blueprint3.RadioGroup-0"
                            type="radio"
                            value="ASG"
                          />
                          <span
                            class="bp3-control-indicator"
                          />
                          Auto scaling groups
                        </label>
                        <label
                          class="bp3-control bp3-radio"
                        >
                          <input
                            name="Blueprint3.RadioGroup-0"
                            type="radio"
                            value="ECS"
                          />
                          <span
                            class="bp3-control-indicator"
                          />
                          ECS Service
                        </label>
                        <label
                          class="bp3-control bp3-radio"
                        >
                          <input
                            name="Blueprint3.RadioGroup-0"
                            type="radio"
                            value="RDS"
                          />
                          <span
                            class="bp3-control-indicator"
                          />
                          RDS instances
                        </label>
                      </div>
                      <div
                        class="Layout--vertical StyledProps--main"
                        style="margin-bottom: 20px;"
                      >
                        <p
                          class="StyledProps--font StyledProps--main"
                          style="font-size: 14px; color: rgb(146, 147, 171); margin-bottom: 10px;"
                        >
                          ce.co.autoStoppingRule.configuration.step2.additionalResourceInfo.kubernetes
                        </p>
                        <p
                          class="StyledProps--font StyledProps--main"
                          style="cursor: pointer; color: rgb(2, 120, 213);"
                        >
                          + ce.co.autoStoppingRule.configuration.step2.addResourceCta.kubernetes
                        </p>
                      </div>
                    </div>
                  </div>
                </div>
                <div
                  class="Layout--vertical Layout--layout-spacing-small StyledProps--main configStepContainer"
                  id="configStep4"
                >
                  <div
                    class="Layout--horizontal StyledProps--main stepInfo"
                  >
                    <span
                      style="color: rgb(89, 43, 170); background-color: rgb(225, 208, 255);"
                    >
                      Step 3 of 3
                    </span>
                  </div>
                  <p
                    class="StyledProps--font StyledProps--main title"
                    data-tooltip-id="awsSetupAdvancedConfig"
                  >
                    ce.co.autoStoppingRule.configuration.step4.setup ce.co.autoStoppingRule.configuration.step4.advancedConfiguration
                  </p>
                  <p
                    class="StyledProps--font StyledProps--main subTitle"
                  >
                    ce.co.gatewayConfig.advancedConfigDescription
                  </p>
                  <div
                    class="childrenContainer"
                  >
                    <div
                      class="Layout--vertical Layout--layout-spacing-medium StyledProps--main"
                    >
                      <label
                        class="Toggle--toggle StyledProps--main"
                        data-testid="progressPageViewToggle"
                      >
                        <input
                          class="Toggle--input"
                          type="checkbox"
                          value=""
                        />
                        <span
                          class="Toggle--toggleIcon"
                        />
                        Hide Progress Page
                      </label>
                      <div
                        class="bp3-tabs Tabs--main Tabs--horizontal"
                      >
                        <div
                          class="bp3-tab-list"
                          role="tablist"
                        >
                          <div
                            class="bp3-tab-indicator-wrapper"
                            style="height: 0px; transform: translateX(0px) translateY(0px); width: 0px;"
                          >
                            <div
                              class="bp3-tab-indicator"
                            />
                          </div>
                          <div
                            aria-controls="bp3-tab-panel_advancedConfigTabs_deps"
                            aria-disabled="false"
                            aria-expanded="true"
                            aria-selected="true"
                            class="bp3-tab"
                            data-tab-id="deps"
                            id="bp3-tab-title_advancedConfigTabs_deps"
                            role="tab"
                            tabindex="0"
                          >
                            ce.co.autoStoppingRule.configuration.step4.tabs.deps.title
                          </div>
                          <div
                            aria-controls="bp3-tab-panel_advancedConfigTabs_schedules"
                            aria-disabled="false"
                            aria-expanded="false"
                            aria-selected="false"
                            class="bp3-tab"
                            data-tab-id="schedules"
                            id="bp3-tab-title_advancedConfigTabs_schedules"
                            role="tab"
                            tabindex="0"
                          >
                            ce.co.autoStoppingRule.configuration.step4.tabs.schedules.title
                          </div>
                        </div>
                        <div
                          aria-hidden="false"
                          aria-labelledby="bp3-tab-title_advancedConfigTabs_deps"
                          class="bp3-tab-panel"
                          id="bp3-tab-panel_advancedConfigTabs_deps"
                          role="tabpanel"
                        >
                          <div
                            class="Layout--vertical Layout--layout-spacing-medium StyledProps--main"
                          >
                            <p
                              class="StyledProps--font StyledProps--main"
                            >
                              ce.co.autoStoppingRule.configuration.step4.tabs.deps.description
                            </p>
                            <button
                              class="bp3-button Button--button StyledProps--font StyledProps--main Button--with-current-color Button--withLeftIcon"
                              style="max-width: 180px;"
                              type="button"
                            >
                              <span
                                class="bp3-icon bp3-icon-plus StyledProps--main"
                                icon="plus"
                              >
                                <svg
                                  data-icon="plus"
                                  height="16"
                                  viewBox="0 0 16 16"
                                  width="16"
                                >
                                  <desc>
                                    plus
                                  </desc>
                                  <path
                                    d="M13 7H9V3c0-.55-.45-1-1-1s-1 .45-1 1v4H3c-.55 0-1 .45-1 1s.45 1 1 1h4v4c0 .55.45 1 1 1s1-.45 1-1V9h4c.55 0 1-.45 1-1s-.45-1-1-1z"
                                    fill-rule="evenodd"
                                  />
                                </svg>
                              </span>
                              <span
                                class="bp3-button-text"
                              >
                                 Add dependency
                              </span>
                            </button>
                          </div>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="Layout--horizontal Layout--layout-spacing-medium StyledProps--main footer"
    >
      <button
        class="bp3-button bp3-disabled Button--button StyledProps--font StyledProps--main Button--with-current-color Button--withLeftIcon"
        disabled=""
        tabindex="-1"
        type="button"
      >
        <span
          class="bp3-icon bp3-icon-chevron-left StyledProps--main StyledProps--padding-right-xsmall"
          icon="chevron-left"
        >
          <svg
            data-icon="chevron-left"
            height="16"
            viewBox="0 0 16 16"
            width="16"
          >
            <desc>
              chevron-left
            </desc>
            <path
              d="M7.41 8l3.29-3.29c.19-.18.3-.43.3-.71a1.003 1.003 0 00-1.71-.71l-4 4C5.11 7.47 5 7.72 5 8c0 .28.11.53.29.71l4 4a1.003 1.003 0 001.42-1.42L7.41 8z"
              fill-rule="evenodd"
            />
          </svg>
        </span>
        <span
          class="bp3-button-text"
        >
          Previous
        </span>
      </button>
      <button
        class="bp3-button Button--button StyledProps--font StyledProps--main StyledProps--intent-primary Button--with-current-color Button--withLeftIcon"
        type="button"
      >
        <span
          class="bp3-icon bp3-icon-chevron-right StyledProps--main StyledProps--padding-right-xsmall"
          icon="chevron-right"
        >
          <svg
            data-icon="chevron-right"
            height="16"
            viewBox="0 0 16 16"
            width="16"
          >
            <desc>
              chevron-right
            </desc>
            <path
              d="M10.71 7.29l-4-4a1.003 1.003 0 00-1.42 1.42L8.59 8 5.3 11.29c-.19.18-.3.43-.3.71a1.003 1.003 0 001.71.71l4-4c.18-.18.29-.43.29-.71 0-.28-.11-.53-.29-.71z"
              fill-rule="evenodd"
            />
          </svg>
        </span>
        <span
          class="bp3-button-text"
        >
          next
        </span>
      </button>
    </div>
  </div>
</div>
`;
